File: 1. Algorithmic Toolbox/Week 3/Largest number problem/largest.java
import java.util.Scanner;
class largest {
    public static void main(String[] args) {
        System.out.println("Enter the number of digits");
        Scanner sc = new Scanner(System.in);
        int num,n = sc.nextInt();
        int arr[] = new int[n];
        for (int i = 0;i < n; i++) {
            arr[i] = sc.nextInt();
        }
        num = 0;
        for (int i = 0;i < n ; i++) {
            int ind = larg(arr, n);
            num = (num*10) + arr[ind];
            arr[ind] = -1;
        }
        System.out.println("maximum number = " + num);
        sc.close();
    }
    public static int larg(int arr[], int n) {
        int ind = 0;
        int max = -1;
        for(int i = 0; i < n; i++) {
            if (arr[i] > max) {
                max = arr[i];
                ind = i;
            }
        }
        return ind;
    }
}

File: 1. Algorithmic Toolbox/Week 3/Largest number problem/largest.c
#include<stdio.h>
int larg(int arr[], int n);
int main() {
    printf("enter the number of digits\n");
    int i,n,num=0;
    scanf("%d",&n);
    int arr[n];
    printf("enter the digits\n");
    for (i=0;i<n;i++) {
        scanf("%d",&arr[i]);
    }
    for (i=0;i<n;i++) {
        int index = larg(arr, n);
        num = (num*10) + arr[index];
        arr[index] = -1;
    }
    printf("maximum number = %d",num);
    return 0;
}
int larg(int arr[], int n) {
    int index,i,max = -1;
    for(i = 0; i < n; i++) {
        if (arr[i] > max) {
            max = arr[i];
            index = i;
        }
    }
    return index;
}